vm-run

```diff
@@ -160,6 +160,9 @@ try:
     # execute any -e and -s commands
     if args.execute:
         for cmd in args.execute:
+            if cmd == 'poweroff -f':
+                machine.kill()
```

Comment:
I wonder about this special case -- what goes wrong with running this directly through `execute()`? We often do this through our tests, even though we usually do some hacks like `m.spawn('sleep 1; poweroff')`. This is a bit too magic for my taste, I'd rather add a `--power-off` option  to it.

But more importantly, why don't you use bots/image-customize for this? This is meant for this kind of sort-running automated/batched processes. We use it all the time in starter-kit and friends.

    ./image-customize -v -r 'rpm -qa' fedora-33

That *almost* gives you the right result -- you need `-v` to show stdout, and then you also get some `set -x` output. The latter goes to stderr, so that's harmless; the only thing you need to fix is to print the machine's UUID to stderr instead of stdout.

Reply:
I got rid of this and followed your suggestion to use the python API.  Thanks for this idea, it's obviously *way* better.